### PR TITLE
Normalize CRAM reference bases and throw on MD5 validation failure.

### PR DESCRIPTION
**Pull request by cmnbroad**
### Description

Replacement for https://github.com/samtools/htsjdk/pull/644. Fixes both https://github.com/samtools/htsjdk/issues/627 and https://github.com/samtools/htsjdk/issues/645.

### Checklist

- [x] Code compiles correctly
- [ ] New tests covering changes and new functionality
- [x] All tests passing
- [x] Extended the README / documentation, if necessary
- [x] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

**Comment by droazen** [comment]
@cmnbroad Could you add a unit test for this change?

**Comment by droazen** [comment]
Back to @cmnbroad

File: src/main/java/htsjdk/samtools/cram/build/Utils.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
  */
 package htsjdk.samtools.cram.build;
 
-class Utils {
+public class Utils {
```
Comment (lbergelson) [inline src/main/java/htsjdk/samtools/cram/build/Utils.java]: can you make this class final and add a private default constructor

**Comment by cmnbroad** [comment]
This has a test now, and should ready for merge.

**Comment by yfarjoun** [comment]
👍 merge away!

**Pull request merged**